src/response.ts

```diff
@@ -19,7 +19,7 @@ export interface ResponseParams {
  * @param {Object} responseHeaders, defaults to an empty object ({})
  * @param {Array<Error>} errors, defaults to an empty array ([])
  */
-export class Response {
+export class Response<DataType = unknown> {
```

Comment:
Should DataType be ensured to extend `number | string| boolean | null | object | Array` to match what is actually possible from being returned in JSON.parse?
```suggestion
type ParsedJSON = number | string | boolean | null | Record<string, any> | any[]

export class Response<DataType extends ParsedJSON = unknown> {
```

Reply:
Otherwise I could just plain say that my `.data` method should return a `Date` instance, whereas this is actually not possible.

Reply:
If we did that, then it would not be possible to assign `unknown` as the default, because `Type 'unknown' does not satisfy the constraint 'ParsedJSON'.` And I think `unknown` is a great default, do you agree?

However, going back to your suggestion, we cannot save the situation by adding `unknown` to the union type, lest the whole `ParsedJSON` becomes `unknown`... Which means we have to take `unknown` off the table. 
But if we did, not only would we lose the default but not even users themselves could use `Response<unknown>` and would instead have to specify something (making this a breaking type change).